main/social/personal_data.php

```diff
@@ -0,0 +1,174 @@
+$actions = '';
+
+// LEFT CONTENT
+$socialMenuBlock = '';
+if ($allowSocial) {
+    // Block Social Menu
+    $socialMenuBlock = SocialManager::show_social_menu('personal-data');
+}
+
+//MAIN CONTENT
+$personalDataContent = '<ul>';
+foreach ($properties as $key => $value) {
+    if (is_array($value) || is_object($value)) {
+        //skip in some cases
+        if (!empty($value['date'])) {
+            $personalDataContent .= '<li>'.$key.': '.$value['date'].'</li>';
+        } else {
+            $personalDataContent .= '<li>'.$key.': '.get_lang('ComplexDataNotShown').'</li>';
+        }
+    } else {
+        $personalDataContent .= '<li>'.$key.': '.$value.'</li>';
+    }
+}
+$jsonProperties = json_encode($properties);
+$personalDataContent .= '</ul>';
+
+// Check terms acceptation
+$termsAndConditionsAcceptance = [];
+$termsAndConditionsAcceptance['accepted'] = false;
+if (api_get_setting('allow_terms_conditions') === 'true') {
+    $extraFieldValue = new ExtraFieldValue('user');
+    $value = $extraFieldValue->get_values_by_handler_and_field_variable(
+        $userId,
+        'legal_accept'
+    );
+    $termsAndConditionsAcceptance['icon'] = Display::return_icon('accept_na.png', get_lang('NotAccepted'));
+    if (isset($value['value']) && !empty($value['value'])) {
+        list($legalId, $legalLanguageId, $legalTime) = explode(':', $value['value']);
+        $termsAndConditionsAcceptance['accepted'] = true;
+        $termsAndConditionsAcceptance['icon'] = Display::return_icon('accept.png', get_lang('LegalAgreementAccepted'));
+        $termsAndConditionsAcceptance['date'] = api_get_local_time($legalTime);
+        // @TODO add action handling for button
+        $termsAndConditionsAcceptance['button'] = Display::url(
+            get_lang('DeleteLegal'),
+            api_get_self().'?action=delete_legal&user_id='.$userId,
+            ['class' => 'btn btn-danger btn-xs']
+        );
+    } else {
+        // @TODO add action handling for button
+        $termsAndConditionsAcceptance['button'] = Display::url(
+            get_lang('SendLegal'),
+            api_get_self().'?action=send_legal&user_id='.$userId,
+            ['class' => 'btn btn-primary btn-xs']
+        );
+    }
+    $termsAndConditionsAcceptance['label'] = get_lang('LegalAccepted');
+} else {
+    $termsAndConditionsAcceptance['label'] = get_lang('NoTermsAndConditionsAvailable');
+}
+
+//Build the final array to pass to template
+$personalData = [];
+$personalData['data'] = $personalDataContent;
+$icon = Display::return_icon('export_excel.png', get_lang('Export'), null,ICON_SIZE_MEDIUM);
+$personalData['data_export_icon'] = $icon;
+$personalData['permissions'] = $termsAndConditionsAcceptance;
+$personalData['responsible'] = api_get_setting('personal_data_responsible_org');
+
+$em = Database::getManager();
+/** @var LegalRepository $legalTerms */
+$legalTermsRepo = $em->getRepository('ChamiloCoreBundle:Legal');
+//Get data about the treatment of data
+$treatmentTypes = [
```

Comment:
I'm not sure where to put it either because I'm not sure of the best way to call them. I don't want to have to instantiate an entity just for that. Definitely not there, indeed. I'll let you do that too.